src/apps/activities/api/activities.py

```diff
@@ -157,6 +158,63 @@ async def applet_activities_for_subject(
         return Response(result=result)
 
 
+async def applet_activities_for_target_subject(
+    applet_id: uuid.UUID,
+    subject_id: uuid.UUID,
+    user: User = Depends(get_current_user),
+    language: str = Depends(get_language),
+    session=Depends(get_session),
+    answer_session=Depends(get_answer_session),
+) -> ResponseMulti[ActivityOrFlowWithAssignmentsPublic]:
+    applet_service = AppletService(session, user.id)
+    await applet_service.exist_by_id(applet_id)
+
+    await SubjectsService(session, user.id).exist_by_id(subject_id)
+
+    # Restrict the endpoint access to owners, managers, coordinators, and assigned reviewers
+    await CheckAccessService(session, user.id).check_subject_subject_access(applet_id, subject_id)
+
+    assignments = await ActivityAssignmentService(session).get_all_with_subject_entities(
+        applet_id, QueryParams(filters={"target_subject_id": subject_id})
+    )
+
+    # Only one of these IDs will be `None` at a time, so the resulting type will be a list of UUIDs
+    activity_and_flow_ids_from_assignments: list[uuid.UUID] = [
+        assignment.activity_id or assignment.activity_flow_id  # type: ignore
+        for assignment in assignments
+    ]
+
+    activity_and_flow_ids_from_submissions = await AnswerService(
+        session, user.id, answer_session
+    ).get_activity_and_flow_ids_by_target_subject(subject_id)
+
+    activities_and_flows = await ActivityService(session, user.id).get_activity_and_flow_basic_info_by_ids_or_auto(
+        applet_id, activity_and_flow_ids_from_submissions + activity_and_flow_ids_from_assignments, language
+    )
+
+    result = []
+    for activity_or_flow in activities_and_flows:
+        activity_or_flow_assignments = [
+            assignment
+            for assignment in assignments
+            if assignment.activity_id == activity_or_flow.id or assignment.activity_flow_id == activity_or_flow.id
+        ]
+
+        activity_or_flow.set_status(activity_or_flow_assignments)
+
+        result.append(
+            ActivityOrFlowWithAssignmentsPublic(
+                **activity_or_flow.dict(),
+                assignments=activity_or_flow_assignments,
```

Comment:
The list of assignments used to determine the "active"/"inactive" activity status should not be the list that's filtered by target subject, which is what `activity_or_flow_assignments` contains. Rather, the value for activity status should be unchanged regardless of the target subject context. It's active if it has _any_ assignments (or is auto-assigned), and inactive if not.

However, we do still want the list of assignments returned in the `assignments` property to be the ones associated with the target subject, so we still need the value of this array. But we need to pass another array of assignments, unfiltered by target subject, to the `ActivityOrFlowWithAssignmentsPublic` constructor.

Again, really sorry I didn't catch this sooner… I'm just seeing it as I use the interface, and am seeing activity statuses changing when I view the PDP from the context of different target subjects (and they shouldn't).

Reply:
@farmerpaul no worries, I'm cognisant that you're currently integrating this and I expect that additional things may come up. I figure it's more useful to address them here than to open subsequent PRs.

Your description of the active/inactive status doesn't line up with what's in the AC of [M2-7767](https://mindlogger.atlassian.net/browse/M2-7767) or [M2-7825](https://mindlogger.atlassian.net/browse/M2-7825). Both of them state that the participant whose PDP you're viewing should be the target of at least one of the assignments in order for the activity/flow to be considered active.

> Activity/Flow (Active): Which has been assigned, where the Participant has been assigned as the Subject...
> Activity/Flow (Inactive): Which has been assigned, where the Participant has been assigned as the Subject (both if the Respondent is another Participant OR themselves), results have been submitted about the Participant, and then all Respondents have been unassigned this Activity/Flow

> “Active”: Criteria for this is that at least one Participant is assigned as a Respondent to complete this Activity/Flow about the Participant whose Detail Page and “About Participant” filter you are currently viewing
> “Inactive”: Criteria for this is that there are no Participants currently assigned as a Respondent to complete this Activity/Flow about the Participant whose Detail Page and “About Participant” filter you are currently viewing BUT there has previously been data collected about the Subject (either from a past Active Assignment or a Take Now).

I think we should get alignment from product/design on this definition first before I make this change

Reply:
Ah, it is indeed very specific there. I was going by the assignment status [tooltips](https://www.figma.com/design/6BNx63peNVzrHAF1FoZ78x/Multi-Informant-2024?node-id=11773-120562&t=vY3pYmnpexaQSSVy-4), which defines each assignment status on hover:

- Active: At least one Participant is assigned to complete this Activity or Flow.
- Inactive: No Participants are assigned to complete this Activity or Flow.

Those definitions suggest to me that activity/flow status has no correlation with the currently selected participant. But I think Matt may have authored the AC you quoted, so presumably those are authoritative. I'll bring it up in Slack to confirm.